common/src/main/cpp/extensions/openxr_fb_hand_tracking_aim_extension_wrapper.cpp

```diff
@@ -0,0 +1,186 @@
+		OpenXRExtensionWrapperExtension() {
+	ERR_FAIL_COND_MSG(singleton != nullptr, "An OpenXRFbHandTrackingAimExtensionWrapper singleton already exists.");
+
+	request_extensions[XR_FB_HAND_TRACKING_AIM_EXTENSION_NAME] = &fb_hand_tracking_aim_ext;
+	singleton = this;
+}
+
+OpenXRFbHandTrackingAimExtensionWrapper::~OpenXRFbHandTrackingAimExtensionWrapper() {
+	cleanup();
+}
+
+void OpenXRFbHandTrackingAimExtensionWrapper::_bind_methods() {
+	ClassDB::bind_method(D_METHOD("is_enabled"), &OpenXRFbHandTrackingAimExtensionWrapper::is_enabled);
+}
+
+void OpenXRFbHandTrackingAimExtensionWrapper::cleanup() {
+	XRServer *xr_server = XRServer::get_singleton();
+
+	for (int i = 0; i < Hand::HAND_MAX; i++) {
+		if (xr_server != nullptr) {
+			xr_server->remove_tracker(trackers[i]);
+		}
+		trackers[i].unref();
+	}
+
+	fb_hand_tracking_aim_ext = false;
+}
+
+godot::Dictionary OpenXRFbHandTrackingAimExtensionWrapper::_get_requested_extensions() {
+	godot::Dictionary result;
+	for (auto ext : request_extensions) {
+		godot::String key = ext.first;
+		uint64_t value = reinterpret_cast<uint64_t>(ext.second);
+		result[key] = (godot::Variant)value;
+	}
+	return result;
+}
+
+PackedStringArray OpenXRFbHandTrackingAimExtensionWrapper::_get_suggested_tracker_names() {
+	PackedStringArray arr = PackedStringArray();
+	arr.push_back(TRACKER_NAME_LEFT);
+	arr.push_back(TRACKER_NAME_RIGHT);
+	return arr;
+}
+
+void OpenXRFbHandTrackingAimExtensionWrapper::_on_state_ready() {
+	is_project_setting_enabled = ProjectSettings::get_singleton()->get_setting_with_override("xr/openxr/extensions/hand_tracking_aim");
+	if (!is_project_setting_enabled) {
+		return;
+	}
+
+	XRServer *xr_server = XRServer::get_singleton();
+	if (xr_server == nullptr) {
+		return;
+	}
+
+	trackers[Hand::HAND_LEFT].instantiate();
+	trackers[Hand::HAND_LEFT]->set_tracker_type(XRServer::TRACKER_CONTROLLER);
+	trackers[Hand::HAND_LEFT]->set_tracker_name(TRACKER_NAME_LEFT);
+	trackers[Hand::HAND_LEFT]->set_tracker_desc("FB Aim tracker Left");
+	xr_server->add_tracker(trackers[Hand::HAND_LEFT]);
+
+	trackers[Hand::HAND_RIGHT].instantiate();
+	trackers[Hand::HAND_RIGHT]->set_tracker_type(XRServer::TRACKER_CONTROLLER);
+	trackers[Hand::HAND_RIGHT]->set_tracker_name(TRACKER_NAME_RIGHT);
+	trackers[Hand::HAND_RIGHT]->set_tracker_desc("FB Aim tracker Right");
+	xr_server->add_tracker(trackers[Hand::HAND_RIGHT]);
+}
+
+void OpenXRFbHandTrackingAimExtensionWrapper::_on_instance_destroyed() {
+	cleanup();
+}
+
+uint64_t OpenXRFbHandTrackingAimExtensionWrapper::_set_hand_joint_locations_and_get_next_pointer(int32_t p_hand_index, void *p_next_pointer) {
+	ERR_FAIL_INDEX_V_MSG(p_hand_index, Hand::HAND_MAX, reinterpret_cast<uint64_t>(p_next_pointer), vformat("Invalid hand index %d", p_hand_index));
+
+	if (!fb_hand_tracking_aim_ext) {
+		return reinterpret_cast<uint64_t>(p_next_pointer);
+	}
+
+	aim_state[p_hand_index].type = XR_TYPE_HAND_TRACKING_AIM_STATE_FB;
```

Comment:
Let's switch this to brace initialization as discussed on https://github.com/GodotVR/godot_openxr_vendors/pull/100

Reply:
Updated in latest push